src/rgw/rgw_lua.cc

```diff
@@ -175,42 +184,69 @@ int install_packages(const DoutPrefixProvider *dpp, rgw::sal::Driver* driver,
     return 0;
   }
   if (ret < 0) {
-    output.append("failed to get lua package list");
+    ldpp_dout(dpp, 1) << "Lua ERROR: failed to get package list. error: " << ret << dendl;
     return ret;
   }
   // verify that luarocks exists
   const auto p = bp::search_path("luarocks");
   if (p.empty()) {
-    output.append("failed to find luarocks");
+    ldpp_dout(dpp, 1) << "Lua ERROR: failed to find luarocks" << dendl;
     return -ECHILD;
   }
 
+  // create the luarocks parent directory
+  auto rc = create_directory_p(dpp, luarocks_path);
+  if (rc < 0) {
+    ldpp_dout(dpp, 1) << "Lua ERROR: failed to recreate luarocks directory: " <<
+      luarocks_path << ". error: " << rc << dendl; 
+    return rc;
+  }
+
+  // create a temporary sub-directory to install all luarocks packages
+  std::string tmp_path_template = luarocks_path;// fs::temp_directory_path();
+  tmp_path_template.append("/XXXXXX");
+  const auto tmp_luarocks_path = mkdtemp(tmp_path_template.data());
```

Comment:
what if these paths aren't under `/tmp`? should rgw try to clean stuff up on shutdown?

Reply:
I can add that, but i don't think we can always rely on proper shutdowns
maybe we can do a cleanup on startup?
the drawback would be that if we change the parent directory (conf name changed, rgw name/port changed) then we would not be able to do the cleanup - but would still be better than nothing.

Reply:
this is addressed here: 00b7ef69b0017cd84676de3f479df19f695506c4
note that the cleanup is done both on shutdown and startup (to cover the case where radosgw crashed etc.).
this does not cover cases where the luarocks parent install dir changed.
any temporary directories and files in the old luarocks parent install dir will not be cleared.

Reply:
@cbodley could you please have a look at the latest code?